pvac_seq/generate_variant_sequences.py

```diff
@@ -0,0 +1,81 @@
+import argparse
+import csv
+import re
+import sys
+
+#This subroutine is a bit funky but it was designed that way to mirror
+#distance_from_end to increase code readability from the caller's perspective
+def distance_from_start(position, string):
+    return position
+
+def distance_from_end(position, string):
+    return len(string) - 1 - position;
+
+def get_wildtype_subsequence_for_printing(position, wildtype_sequence, peptide_sequence_length, line):
+    chromosome = line[0]
+    start      = line[1]
+    stop       = line[2]
+    #If the wildtype sequence is shorter than the desired peptide sequence
+    #length we use the wildtype sequence length instead so that the extraction
+    #algorithm below works correctly
+    if len(wildtype_sequence) < peptide_sequence_length:
+        peptide_sequence_length = len(wildtype_sequence)
+        print("Wildtype sequence length is shorter than desired peptide sequence length at position (%s, %s, %s). Using wildtype sequence length (%s) instead." % chromosome, start, stop, peptide_sequence_length)
+
+    # We want to extract a subset from @arr_wildtype_sequence that is
+    # $peptide_sequence_length long so that the $position ends
+    # up in the middle of the extracted sequence.
+    # If the $position is too far toward the beginning or end of
+    # @arr_wildtype_sequence there aren't enough amino acids on one side
+    # to achieve this.
+    wildtype_subsequence = str
```

Comment:
This is actually assigning the class str to wildtype_subsequence
To assign an empty string, use a literal `wildtype_subsequence = ""` or instantiate the class `wildtype_subsequence = str()`
The same goes for mutation_position below